metals/src/main/scala/scala/meta/internal/metals/ConfiguredLanguageClient.scala

```diff
@@ -82,6 +82,15 @@ final class ConfiguredLanguageClient(
     }
   }
 
+  override def refreshModel(): CompletableFuture[Unit] = {
+    if (clientConfig.codeLenseRefreshSupport)
+      this.refreshCodeLenses.thenApply(_ => ())
+    else {
+      val params = ClientCommands.RefreshModel.toExecuteCommandParams()
+      CompletableFuture.completedFuture(metalsExecuteClientCommand(params))
```

Comment:
I think we should still use `underlying` instance here instead of `this`. 
As far as I remember:
- the most depth language client is created in `Main.scala` - `val clientProxy = launcher.getRemoteProxy`
- then it's passed to the `server.connectToLanguageClient(clientProxy)`
- `MetalsLanguaseServer` has already `DelegatingLanguageClient` created with NoopLC, but `connectToLanguageClient` changes that

So finally it should looks like:
clientProxy <- `ConfiguredLanguageClient` <- `DelegatingLanguageClient` <- `MetalsLanguageServer`

Reply:
If we do `underlying.metalsExecuteClientCommand` we skip the check that is bellow (`clientConfig.isExecuteClientCommandProvider`). 

BTW shouldn't that check now be done in here ?

Reply:
I think we are both right and wrong at the same time hah :D
`refreshCodeLenses` should be called on underlying since it comes from `lsp.LanguageClient` and `metalsExecuteClientCommand` should be called on this, just as you wrote.

Yeah, since code lenses are implemented as a debug request, that check should be moved here

Reply:
I moved the check now but `refreshCodeLenses` should be independent of debug provider.